src/Maestro/DependencyUpdateErrorProcessor/DependencyUpdateErrorProcessor.cs

```diff
@@ -0,0 +1,227 @@
+           
+                bool.TryParse(DependencyUpdateErrorProcessorOptions.DynamicConfigs["FeatureManagement:DependencyUpdateErrorProcessor"], 
+                    out var autoDependencyUpdateErrorProcessor);
+                if (autoDependencyUpdateErrorProcessor)
+                {
+                    IReliableDictionary<string, DateTime> update =
+                        await StateManager.GetOrAddAsync<IReliableDictionary<string, DateTime>>("update");
+                    try
+                    {
+                        using (ITransaction tx = StateManager.CreateTransaction())
+                        {
+                            var previousTransactionDateTime = await update.GetOrAddAsync(
+                             tx,
+                             "update",
+                             DateTime.Now
+                             );
+                            var updatedTime = await CheckForError(previousTransactionDateTime, DependencyUpdateErrorProcessorOptions.GithubUrl);
+                            await update.TryUpdateAsync(
+                                tx,
+                                "update",
+                                updatedTime.UtcDateTime,
+                                previousTransactionDateTime
+                                );
+                            await tx.CommitAsync();
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        Logger.LogError(ex, "Processing update dependency messages");
+                    }
+                }
+            }
+            else
+            {
+                Logger.LogInformation("Dependency Update Error processor is disabled because no App Configuration was available.");
+            }
+        }
+
+        public async Task<DateTimeOffset> CheckForError(DateTimeOffset previousTransactionDateTime , string whereToCreateIssue)
+        {
+            List<RepositoryBranchUpdateHistoryEntry> list = (from repo in Context.RepositoryBranchUpdateHistory
+                                                             where repo.Success == false
+                                                             where repo.Timestamp > previousTransactionDateTime.UtcDateTime
+                                                             orderby repo.Timestamp ascending
+                                                             select repo).ToList();
+            if (!list.Any())
+            {
+                return previousTransactionDateTime;
+            }
+
+            Logger.LogInformation($"Going to create the github issue");
+            try
+            {
+                await CreateGitHubIssueAsync(previousTransactionDateTime.UtcDateTime,  list[0].Repository, list[0].Branch, 
+                    list[0].Action, list[0].ErrorMessage, list[0].Method , whereToCreateIssue);
+            }
+            catch (Exception ex)
```

Comment:
We are doing a LOT of exception swallowing in this code, where the exceptions are logged and not rethrown... so a lot of these methods just silently don't do what you ask them to do... they just write some error and return, and then we continue on without any way to know that.

Only the very highest level of the loop should be catching the exception, everything else should let them flow.

There should be a _single_ catch block in this entire PR, and the appropriate level that we don't get stuck in an infinitely loop if a particular update isn't working, but is otherwise as high as possible.